src/types/ScreenInfoUiaProvider.cpp

```diff
@@ -605,18 +630,35 @@ IFACEMETHODIMP ScreenInfoUiaProvider::get_SupportedTextSelection(_Out_ Supported
 
 const COORD ScreenInfoUiaProvider::_getScreenBufferCoords() const
 {
-    const CONSOLE_INFORMATION& gci = ServiceLocator::LocateGlobals().getConsoleInformation();
-    return gci.GetScreenBufferSize();
+    return _getTextBuffer().GetSize().Dimensions();
+}
+
+const TextBuffer& ScreenInfoUiaProvider::_getTextBuffer() const
+{
+    return _pData->GetTextBuffer();
+}
+
+const Viewport ScreenInfoUiaProvider::_getViewport() const
+{
+    return _pData->GetViewport();
+}
+
+void ScreenInfoUiaProvider::_LockConsole() noexcept
+{
+    _pData->LockConsole();

```

Comment:
small danger:
`::LockConsole()` != `gci.LockConsole()`. The gci one will dispatch control events (ctrl+C) in unlock, while the global one won't. 

`RenderData` calls the global one, while this used to call the gci one. 

This might be okay, since I don't think people are going to be writing Ctrl+C (as an interrupt) to the console via the UIA provider, but if you were for some reason, be aware it might not actually get dispatched until a `gci.Unlock()`.

Reply:
So, to fix this in the long run, does that mean we need to change how Lock/Unlock works in the `Terminal` class?

Reply:
The whole "dispatching events on unlock" thing needs to be revisited completely at some point. I just filed #2141 for that.